src/rest/AnalyticsEndpoint.ts

```diff
@@ -115,7 +115,7 @@ export class AnalyticsEndpoint {
       return results.data;
     } catch (error) {
       AnalyticsEndpoint.pendingRequest = null;
-      if (this.options.accessToken.isExpired(error)) {
+      if (this.isAnalyticsTokenExpired(error)) {
```

Comment:
Due to the looser condition of code `400` and type `InvalidToken`, and the recursive call on line 121, is there a chance of falling into an infinite loop if someone configures an invalid token rather than an expired one?

If so, should we keep track of the number of renew attempts, and limit it to some number?

Reply:
Yes! there is a maximum of 5 renewal attempts already.